resolwe_bio/processes/support_processors/prepare_geo.yml

```diff
@@ -0,0 +1,267 @@
+# ===========
+# Prepare GEO
+# ===========
+---
+
+- slug: prepare-geo-chipseq
+  name: Prepare GEO - ChIP-Seq
+  requirements:
+    expression-engine: jinja
+    resources:
+      cores: 1
+    executor:
+      docker:
+        image: resolwebio/base:ubuntu-17.10
+  data_name: 'Prepare GEO ({{name}} - ChIP-Seq)'
+  version: 0.0.1
+  type: data:other:geo:chipseq
+  category: analyses
+  persistence: CACHED
+  description: >
+    Prepare ChIP-Seq data for GEO upload.
+  input:
+    - name: reads
+      label: Reads
+      type: list:data:reads:fastq
+      description: >
+        List of reads objects. Fastq files will be used.
+    - name: macs14
+      label: Macs14
+      type: list:data:chipseq:macs14
+      description: >
+        List of macs14 objects. Wiggle files will be used.
+    - name: relations
+      label: Relations
+      type: list:basic:string
+      description: >
+        Sample to background mapper. Accepted format <sample name>:<background sample name>.
+    - name: name
+      label: Collection name
+      type: basic:string
+  output:
+    - name: tarball
+      label: GEO folder
+      type: basic:file
+    - name: table
+      label: Annotation table
+      type: basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      NAME={{ name }}_ChIP-Seq
+      mkdir "${NAME}"
+
+      {% set macs_first = macs14|first %}
+
+      declare -A SAMPLE_TO_PAIRED
+      declare -A SAMPLE_TO_BACKGROUND
+
+      {% for relation in relations %}
+        SAMPLE_TO_BACKGROUND[{{ relation.split(':')[0].split('.')[0] }}]={{ relation.split(':')[1].split('.')[0] }}
```

Comment:
Although Jinja doesn't support dict comprehension and our platform doesn't support the `do` Jinja extension, you can still use the last answer [here](https://stackoverflow.com/questions/36886650/how-to-add-a-new-entry-into-a-dictionary-object-while-using-jinja2). The same suggestion applies to `SAMPLE_TO_PAIRED`.

Reply:
This would become: `{% set x=sample_to_background.__setitem__(relation.split(':')[0].split('.')[0], relation.split(':')[1].split('.')[0]) %}`. I don't see why this is better. @mstajdohar, any thoughts? Most of the code is in bash now. This change would require almost everything to be rewritten into jinja. Some important parts however cannot be rewritten into jinja and would still stay in bash. So we would go from bash heavy code (with some jinja) to jinja heavy code (with some bash). Is that really better and why?

Reply:
This was just a suggestion. I don't object merging this PR as it is.